### PR TITLE
[C-1738] Upgrade react-native to 0.68.5

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Upgrades react-native to 0.68.5, a minor version from 0.67.5

Upgrades various packages that were not playing well with new version:
```
    "react-native-push-notification": "8.1.1",
    "react-native-radial-gradient": "1.1.2",
    "react-native-reanimated": "2.13.0",
```

Replaces broken react-native-scrypt with react-native-fast-crypto
Patches broken deprecated-prop-types warnings

### CODE REVIEW

File: packages/mobile/patches/react-native+0.66.3.patch [hunk 1]
```diff
@@ -1,13 +0,0 @@
-diff --git a/node_modules/react-native/Libraries/Animated/AnimatedEvent.js b/node_modules/react-native/Libraries/Animated/AnimatedEvent.js
-index b6de725..dfb9777 100644
---- a/node_modules/react-native/Libraries/Animated/AnimatedEvent.js
-+++ b/node_modules/react-native/Libraries/Animated/AnimatedEvent.js
-@@ -227,7 +227,7 @@ class AnimatedEvent {
-   }
- 
-   _callListeners(...args: any) {
--    this._listeners.forEach(listener => listener(...args));
-+    this._listeners.forEach(listener => listener?.(...args));
```
Comment (sliptype) [inline packages/mobile/patches/react-native+0.66.3.patch:10]: Was this fixed in react-native 0.68? So we don't need the patch anymore?

File: packages/mobile/android/app/build.gradle [hunk 2]
```diff
@@ -283,7 +283,6 @@ dependencies {
     } else {
         implementation jscFlavor
     }
-    compile project(':react-native-video')
```
Comment (sliptype) [inline packages/mobile/android/app/build.gradle:286]: We sure we don't need this anymore?

**Review by sliptype — APPROVED** [review]
Thank you! Are you applying these changes via the upgrade helper or are you using codemods of some sort?

**Comment by dylanjeffers** [comment]
> Thank you! Are you applying these changes via the upgrade helper or are you using codemods of some sort?

Via the upgrade helper and then updating things as the builds break. Its fairly manual unfortunately

File: packages/mobile/android/app/build.gradle (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/android/app/build.gradle:286]: yeah it was complaining, and looking at react-native-video docs, it doesn't say to do this anymore!

File: packages/mobile/patches/react-native+0.66.3.patch (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/patches/react-native+0.66.3.patch:10]: yeah i wasnt seeing this issue happen in my local testing anymore

**Pull request merged**